File: app/src/main/java/com/example/myanimedata/api/CharacterResponseDetail.java
package com.example.myanimedata.api;

import com.google.gson.annotations.SerializedName;

public class CharacterResponseDetail {
    @SerializedName("data")
    private final CharacterDetail characterDetail;

    public CharacterDetail getCharacterDetail() {
        return characterDetail;
    }

    public CharacterResponseDetail(CharacterDetail characterDetail) {
        this.characterDetail = characterDetail;
    }
}

File: app/src/main/java/com/example/myanimedata/api/CharacterResponseDetail.kt
package com.example.myanimedata.api

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.parcelize.Parcelize

@Parcelize
data class CharacterResponseDetail(
    @field:SerializedName("data") val characterDetail: CharacterDetail? = null
) : Parcelable